lib/client/doc.js

```diff
@@ -648,11 +659,13 @@ Doc.prototype._otApply = function(op, source) {
 
   if (op.create) {
     this._setType(op.create.type);
-    this.data = (this.type.deserialize) ?
-      (this.type.createDeserialized) ?
-        this.type.createDeserialized(op.create.data) :
-        this.type.deserialize(this.type.create(op.create.data)) :
-      this.type.create(op.create.data);
+    this._setData(
+      (this.type.deserialize) ?
+        (this.type.createDeserialized) ?
+          this.type.createDeserialized(op.create.data) :
+          this.type.deserialize(this.type.create(op.create.data)) :
+        this.type.create(op.create.data)
+    );
```

Comment:
Can we change this to ifs as it looks more readable 

```suggestion
let data;

if (this.type.deserialize) {
     if (this.type.createDeserialized) {
       data = this.type.createDeserialized(op.create.data)
     } else {
       data = this.type.deserialize(this.type.create(op.create.data))
     }
} else {
     data = this.type.create(op.create.data)
}
this._setData(data);
```